betty/asyncio.py

```diff
@@ -78,3 +80,20 @@ async def run(self) -> None:
             self._return_value = await self._awaitable
         except BaseException as e:
             self._e = e
+
+
+class ThreadSafeLock:
```

Comment:
Actually, a thread lock also works across async, so we can remove the asyncio.Lock from this class.